test/microbench/org/apache/cassandra/test/microbench/DeletionTimeDeSerBench.java

```diff
@@ -0,0 +1,366 @@
+    String liveDTPcParam;
+
+    final static String RAMParam = "RAM";
+    final static String diskParam = "Disk";
+    @Param({ RAMParam,  diskParam })
+    String diskRAMParam;
+
+    // Files
+    File serMMapedFile = new File(DeletionTimeDeSerBench.class + "_Sermmap");
+    ByteBuffer serMMap = allocateMmapedByteBuffer(serMMapedFile, BUFFER_SIZE);
+    File deserMMapedFile = new File(DeletionTimeDeSerBench.class + "_Desermmap");
+    ByteBuffer deserMMap = allocateMmapedByteBuffer(deserMMapedFile, BUFFER_SIZE);
+    File diskFile = new File(DeletionTimeDeSerBench.class + "_DTBenchTest");
+
+    @TearDown
+    public void tearDown()
+    {
+        serMMapedFile.delete();
+        deserMMapedFile.delete();
+        diskFile.delete();
+    }
+
+    @Benchmark
+    public void testE2ESerializeDT(final Blackhole bh) throws IOException
+    {
+        ByteBuffer buffer = serMMap;
+        Serializer serializer = getSerializer(sstableParam);
+        ArrayList<DeletionTime> dts = getDTs(liveDTPcParam);
+        
+        try(DataOutputStreamPlus out = getSerOut(diskRAMParam, serMMap))
+        {
+            for (int i = 0, m = dts.size(); i < m; i++)
+                serializer.serialize(dts.get(i), out);
+            bh.consume(out);
+        }
+
+        bh.consume(buffer);
+        bh.consume(serializer);
+        bh.consume(dts);
+        buffer.clear();
+    }
+
+    @Benchmark
+    public void testE2EDeSerializeDT(final Blackhole bh) throws IOException
+    {
+        ByteBuffer buffer = deserMMap;
+        Serializer serializer = getSerializer(sstableParam);
+        ArrayList<DeletionTime> dts = getDTs(liveDTPcParam);
+        
+        try(DataOutputStreamPlus out = getSerOut(diskRAMParam, deserMMap))
+        {
+            for (int i = 0, m = dts.size(); i < m; i++)
+                serializer.serialize(dts.get(i), out);
+            bh.consume(out);
+        }
+
+        DataInputPlus in = getDeSerIn(diskRAMParam, deserMMap);
+        for (int i = 0, m = dts.size(); i < m; i++)
+            serializer.deserialize(in);
+
+        bh.consume(buffer);
+        bh.consume(serializer);
+        bh.consume(dts);
+        bh.consume(in);
+        buffer.clear();
+    }
+
+    /**
+     * Test the impact of the different bit ops and write calls. Against memory only.
+     */
+    @Benchmark
+    public void testRawAlgWrites(final Blackhole bh)
+    {
+        if (diskRAMParam != RAMParam)
+            return;
+
+        ArrayList<DeletionTime> dts = getDTs(liveDTPcParam);
+        ByteBuffer buffer = ByteBuffer.allocate(DT_STD_SIZE);
+
+        if (sstableParam == oaSstableParam)
+        {
+            for (int i = 0, m = dts.size(); i < m; i++)
+            {
+                buffer.clear();
+                if (dts.get(i).equals(DeletionTime.LIVE))
+                    buffer.put((byte) 0b1000_0000);
+                else
+                {
+                    buffer.putLong(Math.abs(random.nextLong()));
```

Comment:
This can in theory be MIN_VALUE, because `abs(MIN_VALUE) == MIN_VALUE`. I would do `random.nextLong() & Long.MAX_VALUE`.
Similarly below.

Reply:
Done in latest commit